tests/ui/missing_const_for_fn/could_be_const.rs

```diff
@@ -0,0 +1,55 @@
+#![warn(clippy::missing_const_for_fn)]
+#![allow(clippy::let_and_return)]
+
+use std::mem::transmute;
+
+struct Game {
+    guess: i32,
+}
+
+impl Game {
+    // Could be const
+    pub fn new() -> Self {
+        Self { guess: 42 }
+    }
+}
+
+// Could be const
+fn one() -> i32 {
+    1
+}
+
+// Could also be const
+fn two() -> i32 {
+    let abc = 2;
+    abc
+}
+
+// TODO: Why can this be const? because it's a zero sized type?
+// There is the `const_string_new` feature, but it seems that this already works in const fns?
+fn string() -> String {
+    String::new()
```

Comment:
This is a false positive. The `is_min_const_fn` function has a few flaws, but it's not a problem in the compiler because we also have the `qualify_consts` pass which runs *in addition to* the `is_min_const_fn` checks.